### ISSUE TITLE
`winget list` does not show 'available source'

### ISSUE THREAD

**Issue by fbehrens** [microsoft/winget-cli#999]
an additional column `available source` was displayed when I ran `winget list` the first time, but not anymore.

I am running windows 10 environment  ![image](https://user-images.githubusercontent.com/13704/119813559-b9df2b00-bee9-11eb-818c-db7a85cf1d50.png) in an enterprise environment.

```
PS C:\Windows\System32> winget --version
# v1.0.11451

winget list | select -First 5
# Name                                       Id                                                       Version
# - --------------------------------------------------------------------------------------------------------------------
# Microsoft Support and Recovery Assistant   339020b868450372                                         17.0.4840.8
```
I consider this as a bug, as `winget list --help` says:

> The list command displays the packages installed on the system, as well as whether an update is available

**Comment by denelon** [comment]
@fbehrens It looks like you are filtering the results. If you run the command and let the entire list display, you should see any packages with a match to a source displayed. To see a source, there must be a match to a manifest in one of your configured sources.

The Windows Package Manager can `uninstall` any packages in Add / Remove Programs, but it can only `upgrade` packages when there is a manifest with a later version available in one of your configured sources.

**Comment by fbehrens** [comment]
@denelon  When I run the command the first time, I could see two more columns, one I could see the `available version`, and one I could see `source` (which was `winget`) (Don't remember the exact names). 

Now I see only

```
PS C:\Windows\System32> winget list
Name                                       Id                                                       Version
---------------------------------------------------------------------------------------------------------------------
Microsoft Support and Recovery Assistant   339020b868450372                                         17.0.4840.8
Global Order Tool                          79d4859593898a7c                                         1.7.0.1
Cisco Webex Meetings                       ActiveTouchMeetingClient                                 Unknown
Adobe AIR                                  Adobe AIR                                                32.0.0.125
Adobe Flash Player 32 NPAPI                Adobe Flash Player NPAPI                                 32.0.0.465
CanonicalGroupLimited.Ubuntu20.04onWindows CanonicalGroupLimited.Ubuntu20.04onWindows_79rhkp1fndgsc 2004.2021.222.0
Cisco AnyConnect Secure Mobility Client    Cisco AnyConnect Secure Mobility Client                  4.6.02074
Citrix Workspace 1912                      CitrixOnlinePluginPackWeb                                19.12.2000.7
Progress Telerik Fiddler                   Fiddler2                                                 5.0.20192.25091
FiddlerCoreAPI                             FiddlerCoreAPI                                           Unknown
GitHub Desktop                             GitHub.GitHubDesktop                                     2.8.2
Git                                        Git.Git                                                  2.31.1
Google Chrome                              Google.Chrome                                            91.0.4472.77
Hive Streaming                             Hive Streaming 20.2.2.1361                               20.2.2.1361
KeePass Password Safe 2.47                 KeePassPasswordSafe2_is1                                 2.47
Lenovo Auto Scroll Utility                 LenovoAutoScrollUtility                                  2.21
McAfee Agent                               McAfeeAgent                                              5.6.6.232
Microsoft Edge                             Microsoft.Edge                                           90.0.818.66
Microsoft Edge Update                      Microsoft Edge Update                                    1.3.143.57
Microsoft Help Viewer 2.2                  Microsoft Help Viewer 2.2                                2.2.23107
Microsoft Help Viewer 2.3                  Microsoft Help Viewer 2.3                                2.3.28107
Microsoft SQL Server 2017                  Microsoft SQL Server SQL2017                             Unknown
Microsoft.BingWeather                      Microsoft.BingWeather_8wekyb3d8bbwe                      4.46.31121.0
Microsoft.DesktopAppInstaller              Microsoft.DesktopAppInstaller_8wekyb3d8bbwe              1.11.11451.0
Microsoft.GetHelp                          Microsoft.GetHelp_8wekyb3d8bbwe                          10.2104.41161.0
Microsoft.Getstarted                       Microsoft.Getstarted_8wekyb3d8bbwe                       10.2.41172.0
Microsoft.HEIFImageExtension               Microsoft.HEIFImageExtension_8wekyb3d8bbwe               1.0.40978.0
Microsoft.LanguageExperiencePackde-de      Microsoft.LanguageExperiencePackde-DE_8wekyb3d8bbwe      18362.35.116.0
Microsoft.MSPaint                          Microsoft.MSPaint_8wekyb3d8bbwe                          6.2105.4017.0
Microsoft.Messaging                        Microsoft.Messaging_8wekyb3d8bbwe                        4.1901.10241.1000
Microsoft.Microsoft3DViewer                Microsoft.Microsoft3DViewer_8wekyb3d8bbwe                7.2105.4012.0
Microsoft.MicrosoftStickyNotes             Microsoft.MicrosoftStickyNotes_8wekyb3d8bbwe             3.8.8.0
Microsoft.MixedReality.Portal              Microsoft.MixedReality.Portal_8wekyb3d8bbwe              2000.21041.1051.0
Microsoft.OneConnect                       Microsoft.OneConnect_8wekyb3d8bbwe                       5.2011.3081.0
Microsoft.People                           Microsoft.People_8wekyb3d8bbwe                           10.1909.12456.0
Microsoft.Print3D                          Microsoft.Print3D_8wekyb3d8bbwe                          3.3.791.0
Microsoft.ScreenSketch                     Microsoft.ScreenSketch_8wekyb3d8bbwe                     10.2008.2277.0
Microsoft.StorePurchaseApp                 Microsoft.StorePurchaseApp_8wekyb3d8bbwe                 12103.1001.8.0
Microsoft.VP9VideoExtensions               Microsoft.VP9VideoExtensions_8wekyb3d8bbwe               1.0.41182.0
Microsoft.Wallet                           Microsoft.Wallet_8wekyb3d8bbwe                           2.4.18324.0
Microsoft.WebMediaExtensions               Microsoft.WebMediaExtensions_8wekyb3d8bbwe               1.0.40831.0
Microsoft.WebpImageExtension               Microsoft.WebpImageExtension_8wekyb3d8bbwe               1.0.41203.0
Microsoft.Whiteboard                       Microsoft.Whiteboard_8wekyb3d8bbwe                       21.10405.5658.0
Microsoft.Windows.Photos                   Microsoft.Windows.Photos_8wekyb3d8bbwe                   2020.20120.4004.0
Microsoft.WindowsAlarms                    Microsoft.WindowsAlarms_8wekyb3d8bbwe                    10.2101.28.0
Microsoft.WindowsCalculator                Microsoft.WindowsCalculator_8wekyb3d8bbwe                10.2103.8.0
Microsoft.WindowsCamera                    Microsoft.WindowsCamera_8wekyb3d8bbwe                    2021.105.10.0
Microsoft.WindowsFeedbackHub               Microsoft.WindowsFeedbackHub_8wekyb3d8bbwe               1.2103.1172.0
Microsoft.WindowsMaps                      Microsoft.WindowsMaps_8wekyb3d8bbwe                      10.2101.9.0
Microsoft.WindowsSoundRecorder             Microsoft.WindowsSoundRecorder_8wekyb3d8bbwe             10.2103.28.0
Microsoft.WindowsStore                     Microsoft.WindowsStore_8wekyb3d8bbwe                     12104.1001.1.0
Windows Terminal Preview                   Microsoft.WindowsTerminalPreview                         1.9.1445.0
Microsoft.Winget.Source                    Microsoft.Winget.Source_8wekyb3d8bbwe                    2021.527.1953.137
Microsoft.YourPhone                        Microsoft.YourPhone_8wekyb3d8bbwe                        1.21042.110.0
Mozilla Firefox 78.9.0 ESR (x86 de)        Mozilla Firefox 78.9.0 ESR (x86 de)                      78.9.0
Mozilla Maintenance Service                MozillaMaintenanceService                                78.9.0
Npcap                                      NpcapInst                                                0.9986
Microsoft 365 Apps for enterprise          Microsoft.Office                                         16.0.13127.21624
Microsoft 365 Apps for enterprise          Microsoft.Office                                         16.0.13127.21624
OBS Studio                                 OBSProject.OBSStudio                                     25.0.8
Microsoft OneDrive                         Microsoft.OneDrive                                       21.073.0411.0002
Microsoft Teams                            Microsoft.Teams                                          1.4.00.11161
USBPcap 1.5.3.0                            USBPcap                                                  1.5.3.0
ThinkPad USB-C Dock USB Audio              VID_17EF&PID_3063&MI_00                                  1.31.38.18
VLC media player                           VideoLAN.VLC                                             3.0.12
Wireshark 3.2.1 64-bit                     Wireshark                                                3.2.1
Visual Studio Enterprise 2019              Microsoft.VisualStudio.2019.Enterprise                   16.3.29318.209
Visual Studio Community 2019               Microsoft.VisualStudio.2019.Community                    16.10.31321.278
microsoft.windowscommunicationsapps        microsoft.windowscommunicationsapps_8wekyb3d8bbwe        16005.13426.2092…
WinSCP                                     WinSCP.WinSCP                                            5.15.5
Microsoft .NET Core SDK 2.1.701 (x64)      {016b678e-a57a-496c-97cb-5d6b7916ed2f}                   2.1.701
IIS 10.0 Express                           {0307C98E-AE82-4A4F-A950-A72FBD805338}                   10.0.04403
Thinkpad USB Ethernet Adapter Driver       {04201224-2B34-4EE7-862B-B7BBF89DB3AB}                   10.24.326.2018
Microsoft .NET Core SDK 2.2.203 (x64)      {04f9c4c9-b2db-4e7c-808a-5a1c63feca5a}                   2.2.203
Microsoft Visual C++ 2013 Redistributable… {050d4fc8-5d48-4b8f-8972-47c82c46020f}                   12.0.30501.0
Skype for Business Online, Windows PowerS… {0708829b-d81c-4855-8e35-743832369b32}                   7.0.1994.0
Cisco AnyConnect Start Before Login Module {0A2F97FE-DDD8-4F15-8A65-1D1FEA68AD35}                   4.6.02074
Microsoft Power BI Desktop (x64)           {145C0676-7941-4F9E-B41A-D2DCE986BB28}                   2.70.5494.761
Microsoft .NET Core SDK 2.1.801 (x64)      {18b64776-7102-480b-8343-68aff0a12602}                   2.1.801
Microsoft .NET Framework 4.5.1 SDK         {19A5926D-66E1-46FC-854D-163AA10A52D3}                   4.5.51641
McAfee Client Proxy                        {1F7D2FAF-9B32-467D-882F-FF62226C5BF7}                   3.2.0.189
Microsoft Visual Studio Code Insiders (Us… {217B4C08-948D-4276-BFBB-BEE930AE5A2C}_is1               1.56.0
7-Zip                                      7zip.7zip                                                19.00.00.0
Microsoft .NET Core SDK 2.1.700 (x64)      {2335da6c-07b8-4479-b89c-f52986f760f6}                   2.1.700
Microsoft SQL Server 2017 Policies         {256EDCB9-A64D-433C-A1DC-C76F02475915}                   14.0.1000.169
Java 8 Update 201                          {26A24AE4-039D-4CA4-87B4-2F32180201F0}                   8.0.2010.9
Java 8 Update 201 (64-bit)                 {26A24AE4-039D-4CA4-87B4-2F64180201F0}                   8.0.2010.9
Microsoft SQL Server 2014 Management Obje… {2774595F-BC2A-4B12-A25B-0C37A37049B0}                   12.0.2000.8
Microsoft Visual C++ 2015-2019 Redistribu… {282975d8-55fe-4991-bbbb-06a72581ce58}                   14.24.28127.4
Microsoft .NET Framework 4.5.2 Multi-Targ… {290FC320-2F5A-329E-8840-C4193BD7A9EE}                   4.5.51209
Microsoft Web Deploy 4.0                   {2EC26D34-FB67-4C58-AC20-235697551222}                   10.0.3802
Update for Microsoft Visual Studio 2015 (… {2e13383e-8f68-4bb5-959e-9fddd8cb8789}                   14.0.23317
Cisco Jabber                               {302F90D6-C290-42C1-951D-D152AE6B1870}                   12.5.1.27406
Microsoft Teams                            Microsoft.Teams                                          1.3.0.13565
Intel® PROSet/Wireless Software            {3c598844-1b8b-41f0-b5b2-bc1dcf4d47ad}                   20.50.0
CrowdStrike Windows Sensor                 {40f0e3fb-3cc9-4bc2-a2ac-af563467638f}                   6.21.13510.0
Softerra LDAP Browser 4.5 (64-bit)         {413EDBFA-6FE0-4416-B5FA-059C93A6A553}                   4.5.19808.0
Azure Data Studio                          Microsoft.AzureDataStudio                                1.28.0
Yarn                                       Yarn.Yarn                                                1.17.3
Microsoft Visual Studio Installer          {6F320B93-EE3C-4826-85E0-ADF79F8D4C61}                   2.10.2174.31177
Microsoft System CLR Types for SQL Server… {718FFB65-F6E4-4D62-861F-ED10ED32C936}                   12.0.2402.11
Microsoft System CLR Types for SQL Server… {725CC962-98BD-42C7-87D8-51C680FB1779}                   15.0.1200.24
Windows 10 Feature Update Rollout Tool     {767CCE6A-AF88-44A0-909B-B21DDB561E55}                   2.0.0.112
Microsoft ODBC Driver 13 for SQL Server    {76CF9EF4-ABA0-484E-8042-12B99499AF5F}                   14.0.1000.169
Visual Studio Code                         Microsoft.VisualStudioCode                               1.56.2
Microsoft .NET SDK 5.0.300 (x64) from Vis… {7D721068-4D31-4A38-B152-A4112C38708E}                   5.3.21.26805
BayerLabelingPPT2010                       {7FFFE37F-5BE6-4046-AFE2-7EA0AFBFD2B0}                   1.1.0001
Microsoft SQL Server Data-Tier Applicatio… {8074FE78-BDB1-4E15-B525-E73C95F4328D}                   14.0.4127.1
GoldDisk Dynamic Application Installer     {817497B0-2CEA-4B5C-855B-512734BFA726}                   3.0.5
Microsoft ODBC Driver 17 for SQL Server    {853997DA-6FCB-4FB9-918E-E0FF881FAF65}                   17.7.2.1
Lenovo Performance Tuner                   {8955517A-76AD-43C2-A63C-0BF05B998713}_is1               2.0.0.0
Microsoft Silverlight                      {89F4137D-6C26-4A84-BDB8-2E5A4BB71E00}                   5.1.50907.0
Windows Subsystem for Linux Update         {8D646799-DB00-4000-AE7A-756A05A4F1D8}                   5.4.72
Microsoft System CLR Types for SQL Server… {8D7CE3B0-5379-46FE-9F4B-A65D9F4CC1F1}                   15.0.1200.24
Microsoft SQL Server 2016 LocalDB          {9097BF1A-13A0-4A4A-A1F8-473E2A669863}                   13.1.4001.0
Microsoft SQL Server Management Studio - … {91a1b895-c621-4038-b34a-01e7affbcb6b}                   14.0.17289.0
Lotus Notes 8.5.3                          {95246D82-99D2-4229-841E-6867C3251087}                   8.53.11258
Microsoft Visual C++ 2012 Redistributable… {95716cce-fc71-413f-8ad5-56c2892d4b3a}                   11.0.60610.1
PuTTY release                              PuTTY.PuTTY                                              0.73.0.0
BayerLabelingWord2013                      {9A8AB5E4-4E5B-4B63-ADD4-60F79020A7F1}                   1.1.0002
Microsoft OLE DB Driver for SQL Server     {9D6F8754-28E9-4940-B319-3FC8588CF18F}                   18.5.0.0
Microsoft System CLR Types for SQL Server… {9D78F5D4-79D2-4FC6-AC56-F364A0ABC54F}                   14.0.1000.169
Microsoft SQL Server 2012 Native Client    {9D93D367-A2CC-4378-BD63-79EF3FE76C78}                   11.4.7462.6
System Information Reporter                {A215D91A-C60B-418C-B897-E8EC6B854E43}                   1.0.1.203
PDF-Viewer                                 {A278382D-4F1B-4D47-9885-8523F7261E8D}_is1               2.5.322.10
Microsoft .NET Framework 4.8 Targeting Pa… {A4EA9EE5-7CFF-4C5F-B159-B9B4E5D2BDE2}                   4.8.03761
PowerShell                                 Microsoft.PowerShell                                     7.1.3.0
ClickOnce Bootstrapper Package for Micros… {A89F4446-3B75-433B-91B3-C88868CA8544}                   4.8.03928
GlobalPKI Client                           {AA16B81F-FCFF-463D-8E2E-09074B79AE41}                   2.4.12
Adobe Acrobat Reader DC                    Adobe.AdobeAcrobatReaderDC                               21.001.20145
ThunderboltTM Software                     {B020654D-0A98-49AD-AC85-CF92823B965F}                   17.4.76.300
Microsoft SQL Server 2017 T-SQL Language … {B15741CE-BA4E-4F04-A378-B15DE4002E26}                   14.0.17289.0
NVIDIA WMI 2.33.0                          {B2FE1952-0186-46C3-BAEC-A80AA35AC5B8}_Display.NVWMI     2.33.0
NVIDIA nView 149.77                        {B2FE1952-0186-46C3-BAEC-A80AA35AC5B8}_Display.NView     149.77
Microsoft Power BI Desktop (May 2019) (x6… {B3F3CD5B-A6A1-4A84-A8AE-B8B1143C67A6}                   2.69.5467.1801
DisplayLink Graphics Driver                {B42FFF03-2403-4AFF-855E-0CF5F02D8F33}                   8.5.3490.0
Node.js                                    OpenJS.Nodejs                                            16.2.0
Microsoft .NET Framework 4.8 Targeting Pa… {BAAF5851-0759-422D-A1E9-90061B597188}                   4.8.03761
CheckBitlockerBackupService                {BCA544CE-7115-4ECE-9431-D51C9AA4615F}                   1.1.0
Microsoft Search in Bing                   {C17F6DEF-D34C-4B75-97E1-D81062408B4A}                   2.0.2
FORCEPOINT ONE ENDPOINT                    {C5897B83-D970-481F-BB2A-C472BC8CB709}                   19.08.4131
Update for  (KB2504637)                    {CFEF48A8-BFB8-3EAC-8BA5-DE4F8AA267CE}.KB2504637         1
Microsoft .NET Framework 4.5.1 Multi-Targ… {D3517C62-68A5-37CF-92F7-93C029A89681}                   4.5.50932
Microsoft ASP.NET MVC 2                    {DD8FF2F3-0D97-4CF3-AF78-FA0E1B242244}                   2.0.60926.0
GoldDisk WLAN Watcher                      {DFC9D853-B383-4C4C-8498-49B5F8B14483}                   01.00.0008
OutlookCheckMailAddin                      {E0F415C9-0084-4293-BF51-79EE01805429}                   1.0.1
Windows SDK AddOn                          {E6F877A1-2F65-4BF0-87B6-A4071B7663D3}                   10.1.0.0
Microsoft Azure Information Protection     {EC66B856-946E-4B66-ACF9-CAF472A16E1A}                   1.41.51.0
Cisco WebEx Meeting Center                 {ED8D5D15-627B-4B83-953B-A82C1A1A9690}                   32.15.33.8
Intel(R) Processor Graphics                {F0E3AD40-2BBD-4360-9C76-B9AC9A5886EA}                   26.20.100.6951
GlobalPKI Service                          {F21BAF83-5ACB-4AE4-896D-2B31602AD4D3}                   2.4.12
MER for ePO                                {F3256CB3-3C73-4B77-9540-F7F5C94BD01F}                   4.1.0.0
ThinkPad Thunderbolt 3 Dock and USB-C Dock {FFC3CA14-EF69-400C-A872-91E8B65FDD43}_is1               1.001.2
Microsoft Visual C++ 2012 Redistributable… {a1909659-0a08-4554-8af1-2175904903a1}                   11.0.60610.1
Microsoft Visual Studio Tools for Applica… {ab213ab7-4792-4c6f-a3fa-8485d06c3475}                   14.0.23829
Microsoft .NET Core SDK 2.1.402 (x64)      {b415bfcd-0c1a-424c-93f3-03fd83fcc44e}                   2.1.402
Microsoft Visual Studio Tools for Applica… {bd4ef7af-dfb1-472e-8fa4-1b97f360a3e7}                   14.0.23107.20
Microsoft SQL Server Management Studio - … {bf0d55ea-f272-49bc-8699-22fbdcc115a8}                   15.0.18384.0
Microsoft .NET Core SDK 2.1.602 (x64)      {ce5d125b-e426-441b-a83f-d6ef6825aa77}                   2.1.602
Microsoft .NET Core SDK 2.1.802 (x64)      {d115381e-8625-4088-9857-e022d8370667}                   2.1.802
Microsoft .NET Core SDK 2.1.604 (x64)      {d198c231-829e-4f4b-832d-0571aa77244a}                   2.1.604
Microsoft Visual Studio 2015 Shell (Isola… {d2981c27-a434-4c9a-96c7-0209e97c4eac}                   14.0.23107.10
Privilege Management for Windows (x64) 21… {d3789061-fd8c-4bfe-8b4b-768ead470ee8}                   21.1.133.0
Microsoft .NET Core SDK 2.1.500 (x64)      {d83984c4-b4ab-41e1-8d62-84f151ca642b}                   2.1.500
Microsoft Visual C++ 2015-2019 Redistribu… {e31cb1a4-76b5-46a5-a084-3fa419e82201}                   14.24.28127.4
Microsoft Visual C++ 2013 Redistributable… {f65db027-aff3-4070-886a-0d87064aabb1}                   12.0.30501.0
Microsoft Visual Studio Tools for Applica… {f895a2f1-ae3f-4212-8af1-7fa1f8c212ea}                   15.0.27520
PS C:\Windows\System32>
```

**Comment by ThioJoe** [comment]
I've noticed when I run "winget list", it lists everything, but will only show "available" and "source" for a package if it believes that package is out of date.

For example, at first when I ran "winget list", and then ran "winget upgrade", all the programs listed from "winget upgrade" were the _only_ ones that had data in the 'available' and 'source' columns from the "winget list" command.

**Comment by denelon** [comment]
@fbehrens that is a new one to me. Will you submit a [feedback](http://aka.ms/winget-feedback) report to get logs for us? Share the link so we can associate it to this bug.

**Comment by fbehrens** [comment]
@denelon I would like to. Unfortunately my `Diagnostic Data`-settings are managed by my organisation, which prevents me from sending `required or optional dignostic data.

**Comment by denelon** [comment]
@fbehrens the client does not render the additional columns if there are no updates to any package mapped to any known source. It is a technique to try and maximize the limited real estate in a command-line output. Maybe we should include those columns even if there are no upgrades available.

**Comment by fbehrens** [comment]
@denelon I am having updates on `Yarn` which were not shown. See following output
```
PS C:\Windows\System32> winget list | sls Yarn

Yarn                                       Yarn.Yarn                                                1.17.3

PS C:\Windows\System32> winget upgrade "Yarn"
Found Yarn [Yarn.Yarn]
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/yarnpkg/yarn/releases/download/v1.22.5/yarn-1.22.5.msi
  ██████████████████████████████  1.56 MB / 1.56 MB
Successfully verified installer hash
Starting package install...
Successfully installed

PS C:\Windows\System32> winget list | sls Yarn

Yarn                                       Yarn.Yarn                                                1.22.5
```

**Comment by denelon** [comment]
@fbehrens

We've published the v1.1 [release candidate](https://github.com/microsoft/winget-cli/releases/tag/v1.1.12653) and updated our [troubleshooting guide](https://github.com/microsoft/winget-cli/tree/master/doc/troubleshooting).

Can you confirm this has been resolved?

Related to:
* #1155 

You might be interested in adding your 👍.
